### PR TITLE
tweaks to schema per feedback from composer team

### PR DESCRIPTION
**Pull request by tomlm**
Give consistent OneOf definitions around enums

### CODE REVIEW

**Comment by a-b-r-o-w-n** [comment]
Converting to draft until I can review in Composer!

File: libraries/Microsoft.Bot.Builder.AI.QnA/Schemas/Microsoft.QnAMakerDialog.schema [hunk 1]
```diff
@@ -95,17 +95,19 @@
...
                         "default",
                         "questionOnly",
                         "autoSuggestQuestion"
-                    ]
+                    ],
+                    "default": "default"
                 },
                 {
-                    "pattern": "^=.*\\S.*"
+                    "$ref": "https://raw.githubusercontent.com/microsoft/botframework-sdk/master/schemas/component/definitions.schema#/definitions/equalsExpression"
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.AI.QnA/Schemas/Microsoft.QnAMakerDialog.schema:108]: Shouldn't this be the proxy address for the Schema that we setup last round?

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Converters/ITemplateActivityConverter.cs [hunk 2]
```diff
@@ -12,16 +15,21 @@ namespace Microsoft.Bot.Builder.Dialogs.Adaptive
     /// <summary>
     /// JsonConverter to load ITemplate&lt;Activity&gt;.
     /// </summary>
-    public class ActivityTemplateConverter : JsonConverter
+    public class ITemplateActivityConverter : InterfaceConverter<ITemplate<Activity>>
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Converters/ITemplateActivityConverter.cs:18]: Having a class that's "I" makes my head hurt. Are there alternatives? TemplateActivityConverterBase? Or similar

**Comment by cleemullins** [comment]
Per discussion with @tomlm, some of these classes should be internal which addresses all my comments.

**Pull request merged**